### PR TITLE
Refine socket-api sample

### PR DESCRIPTION
**Pull request by wenyongh**
Refine the source code of tcp_server.c and tcp_client.c
Add building of iwasm with lib-pthread and libc-wasi enabled
Add sample document
Update iwasm --addr-pool argument description

### CODE REVIEW

File: samples/socket-api/wasm-src/tcp_server.c [hunk 1]
```diff
@@ -84,21 +96,22 @@ main(int argc, char *argv[])
...
     }
 
     for (int i = 0; i < WORKER_NUM; i++) {
         pthread_join(workers[i], NULL);
     }
 
-    printf("[Server] Shuting down ... \n");
+    printf("[Server] Shuting down ..\n");
     shutdown(socket_fd, SHUT_RDWR);
+    close(socket_fd);
```
Comment (lum1n0us) [inline samples/socket-api/wasm-src/tcp_server.c:108]: either `shutdown` or `close`,  both are too much

File: samples/socket-api/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+Start the tcp server, which opens port 1234 and waits for clients to connect.
+```bash
+cd build
+./iwasm --addr-pool=0.0.0.0/15 tcp_server.wasm
+```
+
+Start the tcp client, which connects the server and receives message.
+```bash
+cd build
+./iwasm --addr-pool=0.0.0.0/15 tcp_client.wasm
```
Comment (lum1n0us) [inline samples/socket-api/README.md:38]: *127.0.0.1* is better

File: samples/socket-api/wasm-src/tcp_server.c (hunk 1 above)
Comment (wenyongh) [reply samples/socket-api/wasm-src/tcp_server.c]: Done

File: samples/socket-api/README.md (hunk 2 above)
Comment (wenyongh) [reply samples/socket-api/README.md]: Done

File: samples/socket-api/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+And also file `tcp_server` and `tcp_client` of native version are created.
+
+Note that iwasm is built with libc-wasi and lib-pthread enabled.
+
+## Run workload
+
+Start the tcp server, which opens port 1234 and waits for clients to connect.
+```bash
+cd build
+./iwasm --addr-pool=127.0.0.1/15 tcp_server.wasm
```
Comment (lum1n0us) [inline samples/socket-api/README.md:32]: The server should use 0.0.0.0 to allow it to bind to any address. Partially because the code is use 0.0.0.0.

**Comment by lum1n0us** [comment]
LGTM

**Pull request merged**